maestro-cli/src/main/java/maestro/cli/insights/TestAnalysisManager.kt

```diff
@@ -0,0 +1,142 @@
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.stream.Collectors
+import maestro.cli.api.ApiClient
+import maestro.cli.cloud.CloudInteractor
+import maestro.cli.util.EnvUtils
+import maestro.cli.util.PrintUtils
+import maestro.cli.view.box
+import kotlin.io.path.createDirectories
+import kotlin.io.path.exists
+import kotlin.io.path.readText
+import kotlin.io.path.writeText
+
+data class FlowFiles(
+    val imageFiles: List<Pair<ByteArray, Path>>,
+    val textFiles: List<Pair<ByteArray, Path>>
+)
+
+class TestAnalysisManager(private val apiUrl: String, private val apiKey: String?) {
+    private val apiclient by lazy {
+        ApiClient(apiUrl)
+    }
+
+    fun runAnalysis(debugOutputPath: Path): Int {
+        val flowFiles = processFilesByFlowName(debugOutputPath)
+        if (flowFiles.isEmpty()) {
+            PrintUtils.warn("No screenshots or debug artifacts found for analysis.")
+            return 0;
+        }
+
+        return CloudInteractor(apiclient).analyze(
+            apiKey = apiKey,
+            flowFiles = flowFiles,
+            debugOutputPath = debugOutputPath
+        )
+    }
+
+    private fun processFilesByFlowName(outputPath: Path): List<FlowFiles> {
+        val files = Files.walk(outputPath)
+            .filter(Files::isRegularFile)
+            .collect(Collectors.toList())
+
+        return if (files.isNotEmpty()) {
+            val (imageFiles, textFiles) = getDebugFiles(files)
+            listOf(
+                FlowFiles(
+                    imageFiles = imageFiles,
+                    textFiles = textFiles
+                )
+            )
+        } else {
+            emptyList()
+        }
+    }
+
+    private fun getDebugFiles(files: List<Path>): Pair<List<Pair<ByteArray, Path>>, List<Pair<ByteArray, Path>>> {
+        val imageFiles = mutableListOf<Pair<ByteArray, Path>>()
+        val textFiles = mutableListOf<Pair<ByteArray, Path>>()
+
+        files.forEach { filePath ->
+            val content = Files.readAllBytes(filePath)
+            val fileName = filePath.fileName.toString().lowercase()
+
+            when {
+                fileName.endsWith(".png") || fileName.endsWith(".jpg") || fileName.endsWith(".jpeg") -> {
+                    imageFiles.add(content to filePath)
+                }
+
+                fileName.startsWith("commands") -> {
+                    textFiles.add(content to filePath)
+                }
+
+                fileName == "maestro.log" -> {
+                    textFiles.add(content to filePath)
+                }
+            }
+        }
+
+        return Pair(imageFiles, textFiles)
+    }
+
+    /**
+     * The analytics system for Test Analysis.
+     *  - Uses configuration from $XDG_CONFIG_HOME/maestro/analyze-analytics.json.
+     */
+    companion object Analytics {
```

Comment:
We already have analytics class for this why not use that?

Reply:
Analytics was the inspiration initially but rethinking the purpose of this class I think that the naming is kinda wrong, the use of the analytics file rn is related to our tracking of local data, and the whole purpose of this class is to manage the TestAnalysis Notification on the CLI, we are not tracking any data, do you think it would be better to rename this class to `Notification`?